src/Aspire.Hosting.AWS/CloudFormation/CloudFormationProvisioner.cs

```diff
@@ -177,4 +184,34 @@ private static IAmazonCloudFormation GetCloudFormationClient(ICloudFormationReso
             throw new AWSProvisioningException("Failed to construct AWS CloudFormation service client to provision AWS resources.", e);
         }
     }
+
+    private static readonly Regex s_awsRegionRegex = new("^(us|eu|ap|sa|ca|me|af|il)-\\w+-\\d+$", RegexOptions.Singleline);
```

Comment:
Missed opportunity to use regex source generators

https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0